gobblin-iceberg/src/main/java/org/apache/gobblin/iceberg/writer/IcebergMetadataWriter.java

```diff
@@ -422,6 +480,21 @@ private void computeCandidateSchema(GobblinMetadataChangeEvent gmce, TableIdenti
     }
   }
 
+  /**
+   * Add a partition column to the schema and partition spec
+   * @param table incoming iceberg table
+   * @param fieldName name of partition column
+   * @param type datatype of partition column
+   * @return table with updated schema and partition spec
+   */
+  private Table addPartitionToIcebergTable(Table table, String fieldName, String type) {
+    if(!table.schema().columns().stream().anyMatch(x -> x.name().equalsIgnoreCase(fieldName))) {
+      table.updateSchema().addColumn(fieldName, Types.fromPrimitiveString(type)).commit();
```

Comment:
Are we enforcing string type for all partition columns? So late values will be string type as opposed to int?

Reply:
No. We accept the type of new column via config "iceberg.completeness.add.partition.type"